ibis/backends/duckdb/__init__.py

```diff
@@ -437,9 +437,29 @@ def read_in_memory(
         ir.Table
             The just-registered table
         """
+
+        def _clean_up_string_columns(
+            t: pa.Table | pd.DataFrame,
+        ) -> pa.Table | pd.DataFrame:
+            import pandas as pd
+
+            if not isinstance(t, pd.DataFrame):
+                return t
+
+            from ibis.backends.pandas.client import sch
+
+            schema = sch.infer(t)
+            return t.assign(
+                **{
+                    name: col.map(str, na_action="ignore")
```

Comment:
I think we technically only need to do this for columns that are inferred as `str` type if the input pandas type is an `object` column. The other new pandas str dtypes won't result in this bug, and avoiding the conversation in those cases would be slightly faster. Not sure if the extra complications are worth it though.

I also think we'll want to check for `None` here, rather than blindly calling `str`.

```suggestion
                    name: col.map(lambda x: x if x is None else str(x), na_action="ignore")
```

Reply:
`na_action="ignore"` allows us to elide the `None`